pypeit/core/wavecal/wvutils.py

```diff
@@ -390,49 +396,90 @@ def zerolag_shift_stretch(theta, y1, y2):
 
     """
 
-
     shift, stretch = theta
     y2_corr = shift_and_stretch(y2, shift, stretch)
     # Zero lag correlation
     corr_zero = np.sum(y1*y2_corr)
-    corr_denom = np.sqrt(np.sum(y1*y1)*np.sum(y2*y2))
+    corr_denom = np.sqrt(np.sum(y1*y1)*np.sum(y2_corr*y2_corr))
     corr_norm = corr_zero/corr_denom
+    if corr_denom == 0.0:
+        msgs.error('The shifted and stretched spectrum is zero everywhere. Cross-correlation cannot be performed. There is likely a bug somewhere')
     return -corr_norm
 
-def smooth_ceil_cont(inspec1, smooth, percent_ceil = None, use_raw_arc=False,sigdetect = 10.0, fwhm = 4.0):
-    """ Utility routine to smooth and apply a ceiling to spectra """
 
-    # ToDO can we improve the logic here. Technically if use_raw_arc = True and perecent_ceil=None
-    # we don't need to peak find or continuum subtract, but this makes the code pretty uggly.
+def get_xcorr_arc(inspec1, sigdetect=5.0, sig_ceil=10.0, percent_ceil=50.0, use_raw_arc=False, fwhm = 4.0):
+
+    """  Utility routine to create an synthetic arc spectrum for cross-correlation using the location of the peaks in
+    the input spectrum.
+
+    Args:
+        inspec1 (ndarray):
+          Input spectrum, shape = (nspec,)
+        sigdetect (float, optional, default=3.0):
+          Peak finding threshold for lines that will be used to create the synthetic xcorr_arc
+        sig_ceil (float, optional, default = 10.0):
+          Significance threshold for peaks that will be used to determine the line amplitude clipping threshold.
+          For peaks with significance > sig_ceil, the code will find the amplitude corresponding to
+          perecent_ceil, and this will be the clipping threshold.
+        percent_ceil (float, optional, default=50.0):
+          Upper percentile threshold for thresholding positive and negative values. If set to None, no thresholding
+          will be performed.
+        use_raw_arc (bool, optional):
+          If True, use amplitudes from the raw arc, i.e. do not continuum subtract. Default = False
+        fwhm (float, optional):
+          Fwhm of arc lines. Used for peak finding and to assign a fwhm in the xcorr_arc.
 
-    # Run line detection to get the continuum subtracted arc
+    Returns:
+        xcorr_arc (ndarray):
```

Comment:
can't use the variable name.
and should be numpy.ndarray_

Reply:
I like the variable name and this is the sklearn style which I've been pushing for.

Reply:
Need help with docs.